src/java/org/apache/cassandra/serializers/UTF8Serializer.java

```diff
@@ -57,8 +60,51 @@ static <V> boolean validate(V value, ValueAccessor<V> accessor)
             if (value == null)
                 return false;
 
-            int b = 0;
-            int offset = 0;
+            // perf optimizations:
+            // 1) avoid bymorhic/megamorphic calls via ValueAccessor
+            // 2) use a simplified logic to handle ASCII prefixed String scenario faster
+            if (accessor == ByteArrayAccessor.instance)
+            {
+                byte[] valueAsArray = accessor.toArray(value);
+                for (int i = 0; i < valueAsArray.length; i++)
+                {
+                    if (valueAsArray[i] < 0)
+                        return validateSlowPath(value, accessor, i);
+                }
+                return true;
+            }
+
+            if (accessor == ByteBufferAccessor.instance)
+            {
+                ByteBuffer valueAsBuffer = accessor.toBuffer(value);
+                if (valueAsBuffer.hasArray())
+                {
+                    byte[] valueAsArray = valueAsBuffer.array();
+                    int start = valueAsBuffer.position();
+                    int end = start + valueAsBuffer.remaining();
+                    for (int i = start; i < end; i++)
+                    {
+                        if (valueAsArray[i] < 0)
+                            return validateSlowPath(value, accessor, i - start);
+                    }
+                }
+                return true;
```

Comment:
should we always return true? there will never be no validation when `valueAsBuffer.hasArray()` is false. Should you move the return statement into if block above instead?